lib/mods/replace-get-set.js

```diff
@@ -24,47 +24,62 @@ function isDataProperty(type, propertyName) {
       dataProperties.get(type).includes(propertyName);
 }
 
+/**
+ * @param {Node} node - the current ts-morph AST Node
+ *
+ * @return {boolean} whether the node is a global get
+ */
+function isGlobalGet(node) {
+  return nodeTypes.isPropertyAccessExpression(node) &&
+    !nodeTypes.isCallExpression(node.getParent()) &&
+    isGlobal(node.getExpression());
+}
+
+/**
+ * @param {Node} node - the current ts-morph AST Node
+ *
+ * @return {boolean} whether the node is a global set
+ */
+function isGlobalSet(node) {
+  return nodeTypes.isBinaryExpression(node) &&
+    node.getOperatorToken().getText() === '=' &&
+    nodeTypes.isPropertyAccessExpression(node.getLeft()) &&
+    isGlobal(node.getLeft().getExpression());
+}
+
 /**
  * Replaces getset calls in the current node with the Get-Originals version
  *
  * @param {Node} node - a ts-morph AST Node
  * @param {Object} imports - the current list of imports
  */
 function replaceGlobalGetSetCalls(node, imports) {
-  if (nodeTypes.isPropertyAccessExpression(node) &&
-      !nodeTypes.isCallExpression(node.getParent()) &&
-      isGlobal(node.getExpression())) {
-    const propertyName = node.getSymbol().getEscapedName();
-    const baseTypeName = node.getExpression().getType().getSymbol().getName();
+  let args; let type; let nodeToConsider;
 
-    if (isDataProperty(baseTypeName, propertyName)) {
-      return;
-    }
+  if (isGlobalGet(node)) {
+    nodeToConsider = node;
+    type = 'get';
+  } else if (isGlobalSet(node)) {
+    nodeToConsider = node.getLeft();
+    type = 'set';
+    args = node.getRight().getText();
+  } else {
+    return;
+  }
 
-    const base = node.getExpression().getSymbol().getName();
-    let type;
-    let args;
-    let nodeToReplace;
-    const parent = node.getParent();
+  const propertyName = nodeToConsider.getSymbol().getEscapedName();
+  const moduleName = nodeToConsider.getExpression().getType()
+      .getSymbol().getName();
 
-    // Check if the expression is receiving assignment (set)
-    if (nodeTypes.isBinaryExpression(parent) &&
-        parent.getOperatorToken().getText() === '=' &&
-        node.getParent().getLeft() === node) {
-      type = 'set';
-      args = parent.getRight().getText();
-      nodeToReplace = parent;
-    } else {
-      type = 'get';
-      nodeToReplace = node;
-    }
+  if (isDataProperty(moduleName, propertyName)) {
+    return;
+  }
 
-    const original = getOriginal(propertyName, type, baseTypeName);
+  const baseObject = nodeToConsider.getExpression().getSymbol().getName();
```

Comment:
Does the following tweak work?

```js
  const moduleName = nodeToConsider.getExpression().getType().getSymbol().getName();
  const baseObject = nodeToConsider.getExpression().getText();
  const propertyName = nodeToConsider.getName();
```

In particular I'm not sure why we need to indirect through getSymbol for baseObject (since baseObject is an identifier, so its text should work well), and I think .getName() on property access expressions is just a shorter version.

Reply:
Yeah that works